templates/test/ci/cluster-template-prow-aks-multi-tenancy.yaml

```diff
@@ -101,8 +102,38 @@ metadata:
   name: agentpool1
   namespace: default
 spec:
+  maxPods: 64
```

Comment:
Since we're adding a 3rd pool to test the "no configuration specified" pool config scenario, we may as well add a differnt `maxPods` config value to the 2nd one that now serves to validate the 2nd possible valid `OsDiskType` flavor.